VAMobile/src/utils/auth.ts

```diff
@@ -1,18 +1,372 @@
+ */
+export const isBiometricsPreferred = async (): Promise<boolean> => {
+  try {
+    const value = await AsyncStorage.getItem(BIOMETRICS_STORE_PREF_KEY)
+    return value === AUTH_STORAGE_TYPE.BIOMETRIC
+  } catch (e) {
+    logNonFatalErrorToFirebase(e, `isBiometricsPreferred: Auth Service Error`)
+  }
+
+  return false
+}
+
+/**
+ * Clears auth credentials
+ */
+export const clearStoredAuthCreds = async (): Promise<void> => {
+  await Keychain.resetInternetCredentials(KEYCHAIN_STORAGE_KEY)
+  await AsyncStorage.removeItem(REFRESH_TOKEN_TYPE)
+}
+
+/**
+ * Action to check if this is the first time a user has logged in
+ */
+
+export const checkFirstTimeLogin = async (): Promise<boolean> => {
+  const isFirstLogin = !(await AsyncStorage.getItem(FIRST_LOGIN_COMPLETED_KEY))
+  // On the first sign in, clear any stored credentials from previous installs
+  // In integration tests this will change the behavior and make it inconsistent across runs so return false
+  if (isFirstLogin && !IS_TEST) {
+    await clearStoredAuthCreds()
+    return true
+  }
+  return false
+}
+
+/**
+ * Sets the flag used to determine if this is the first time a user has logged into the app
+ */
+export const completeFirstTimeLogin = async (queryClient: QueryClient) => {
+  await AsyncStorage.setItem(FIRST_LOGIN_COMPLETED_KEY, FIRST_LOGIN_STORAGE_VAL)
+  const userSettings = queryClient.getQueryData(authKeys.settings) as UserAuthSettings
+  queryClient.setQueryData(authKeys.settings, { ...userSettings, firstTimeLogin: false })
+}
+
+export const setBiometricsPreference = async (value: boolean, queryClient: QueryClient) => {
+  await AsyncStorage.setItem(BIOMETRICS_STORE_PREF_KEY, value ? AUTH_STORAGE_TYPE.BIOMETRIC : AUTH_STORAGE_TYPE.NONE)
+  const refreshToken = await retrieveRefreshToken()
+  await saveRefreshToken(refreshToken || '')
+  const userSettings = queryClient.getQueryData(authKeys.settings) as UserAuthSettings
+  queryClient.setQueryData(authKeys.settings, { ...userSettings, shouldStoreWithBiometric: value })
+  await setAnalyticsUserProperty(UserAnalytics.vama_uses_biometric(value))
+}
+
+export const debugResetFirstTimeLogin = async (
+  logout: UseMutateFunction<Response, Error, void, void>,
+  queryClient: QueryClient,
+) => {
+  await AsyncStorage.setItem(FIRST_LOGIN_COMPLETED_KEY, '')
+  await logout()
+  await setBiometricsPreference(false, queryClient)
+  const userSettings = queryClient.getQueryData(authKeys.settings) as UserAuthSettings
+  queryClient.setQueryData(authKeys.settings, { ...userSettings, firstTimeLogin: true })
+}
+
+/**
+ * Sets the flag used to determine if the biometrics preference screen should be displayed
+ */
+export const setDisplayBiometricsPreferenceScreen = (value: boolean, queryClient: QueryClient) => {
+  const userSettings = queryClient.getQueryData(authKeys.settings) as UserAuthSettings
+  queryClient.setQueryData(authKeys.settings, { ...userSettings, displayBiometricsPreferenceScreen: value })
+}
```

Comment:
Agreed, I will address all this in your last comment

Reply:
it's this way due to state management in the App.tsx file being needed to determine which screen to view. I agree it shouldn't be this way and I will need a little time to think about how to best change the whole AuthGuard function so that it behaves correctly.